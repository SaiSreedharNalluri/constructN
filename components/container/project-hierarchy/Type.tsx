import { ChildrenEntity } from "../../../models/IStructure";

export interface RenderTree {
  _id: string;
  name: string;
  children?: RenderTree[];
  isSelected: boolean;
}

export interface SelectLayerContainerProps {
  openselectlayer?: boolean;
}

export interface SelectLayerProps {
  title: string;
  openSelectLayer?: boolean;
  onCloseHandler: () => void;
  treeData: any[];
  getStructureData?: (structure: ChildrenEntity) => void;
}
