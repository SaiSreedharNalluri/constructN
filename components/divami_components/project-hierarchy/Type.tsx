import { ChildrenEntity } from "../../../models/IStructure";

export interface RenderTree {
  _id: string;
  name: string;
  children?: RenderTree[];
  isSelected: boolean;
}

export interface SelectLayerContainerProps {
  openselectlayer?: boolean;
}

export interface SelectLayerProps {
  title: string;
  openSelectLayer?: boolean;
  onCloseHandler: () => void;
  treeData: any[];
  getStructureData?: (structure: ChildrenEntity) => void;
  handleSearch: any;
  handleNodeSelection?: any;
  selectedNodes?: any;
  handleNodeExpand?: any;
  expandedNodes?: any;
}

export interface ProjectHierarchyProps extends SelectLayerProps {
  setHierarchy: any;
  updateBreadcrumbData?: any;
}
