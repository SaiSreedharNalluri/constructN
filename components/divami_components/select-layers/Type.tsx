export interface RenderTree {
  id: string
  name: string
  children?: RenderTree[]
  isSelected: boolean
}

export interface SelectLayerContainerProps {
  openSelectLayer: boolean
}

export interface SelectLayerProps {
  title: string
  openSelectLayer: boolean
  onCloseHandler: () => void
}
