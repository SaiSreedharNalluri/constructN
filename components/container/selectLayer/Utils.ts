import type { RenderTree } from "./Type";

export const getTreeViewDataForLayers = (optionsList: any) => {
  // if (optionsList) {
  //   const treeViewData = Object.keys(optionsList).map((key, index) => {
  //     console.log(key, 'key', optionsList[key], 'optionsList[key]');
  //     return {
  //       id: key + index,
  //       name: key,
  //       isSelected: true,
  //     };
  //   });
  //   return treeViewData;
  // }
  let arr = [];
  let index = 0;

  for (const key in optionsList) {
    arr.push({
      id: `p${index}`,
      name: key,
      isSelected: optionsList[key]?.isSelected,
      children: [],
    });
    index = index + 1;
    if (
      optionsList[key].hasOwnProperty("children") &&
      optionsList[key]["children"]?.length
    ) {
      arr[arr.length - 1].children = optionsList[key].children.map(
        (each: any, childIndex: number) => {
          return {
            ...each,
            id: `p${index}ch${childIndex}`,
            name: each.name,
            isSelected: each.isSelected,
            children: [],
          };
        }
      );
    }
  }
  return arr;
};

export const getSelectedLayers = (
  data: RenderTree[] | any,
  itemsChecked: any = []
) => {
  for (let i = 0; i < data.length; i++) {
    if (data?.length && data[i].isSelected) {
      itemsChecked.push(data[i].name);
    }
    if (data?.length && data[i].children) {
      getSelectedLayers(data[i].children, itemsChecked);
    }
  }
  return itemsChecked;
};

export const handleSelection = (data: RenderTree[] | any, id: string) => {
  for (let i = 0; i < data.length; i++) {
    if (data[i].id === id && data[i]) {
      data[i].isSelected = !data[i].isSelected;
      return data;
    }
    if (data[i].children) {
      data[i].children = handleSelection(data[i].children, id);
    }
  }
  return data;
};
