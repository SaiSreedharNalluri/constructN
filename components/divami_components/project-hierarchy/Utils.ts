import { ChildrenEntity } from '../../../models/IStructure'
import type { RenderTree } from './Type'

export const getSelectedLayers = (
  data: ChildrenEntity[],
  itemsChecked: string[] = [],
) => {
  for (let i = 0; i < data.length; i++) {
    // if (data[i].isSelected) {
    //   itemsChecked.push(data[i].name)
    // }
    if (data[i].children) {
      getSelectedLayers(data[i].children as ChildrenEntity[], itemsChecked)
    }
  }
  return itemsChecked
}
export const getAllIds = (data: any, nodes: string[] = []) => {
  for (let i = 0; i < data.length; i++) {
    nodes.push(data[i]._id)
    if (data[i].children) {
      getAllIds(data[i].children as ChildrenEntity[], nodes)
    }
  }
  return nodes
}
export const handleSelection = (data: RenderTree[], id: string) => {
  for (let i = 0; i < data.length; i++) {
    if (data[i]._id === id && data[i]) {
      data[i].isSelected = !data[i].isSelected
      return data
    }
    if (data[i].children) {
      data[i].children = handleSelection(data[i].children as RenderTree[], id)
    }
  }
  return data
}
