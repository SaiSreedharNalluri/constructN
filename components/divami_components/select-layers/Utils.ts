import type { RenderTree } from "./Type";

export const getSelectedLayers = (
  data: RenderTree[],
  itemsChecked: string[] = []
) => {
  for (let i = 0; i < data.length; i++) {
    if (data?.length && data[i].isSelected) {
      itemsChecked.push(data[i].name);
    }
    if (data?.length && data[i].children) {
      getSelectedLayers(data[i].children, itemsChecked);
    }
  }
  return itemsChecked;
};

export const handleSelection = (data: RenderTree[], id: string) => {
  for (let i = 0; i < data.length; i++) {
    if (data[i].id === id && data[i]) {
      data[i].isSelected = !data[i].isSelected;
      return data;
    }
    if (data[i].children) {
      data[i].children = handleSelection(data[i].children, id);
    }
  }
  return data;
};
