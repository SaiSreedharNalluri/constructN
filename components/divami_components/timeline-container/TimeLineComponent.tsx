import { useEffect, useState } from "react";
import Moment from "moment";
import Pagination from "@mui/material/Pagination";
import PaginationItem from "@mui/material/PaginationItem";
import { KeyboardArrowDown } from "@mui/icons-material";
import KeyboardDoubleArrowLeftIcon from "@mui/icons-material/KeyboardDoubleArrowLeft";
import KeyboardDoubleArrowRightIcon from "@mui/icons-material/KeyboardDoubleArrowRight";
import { ISnapshot } from "../../../models/ISnapshot";
import CustomCalender from "../custom-datepicker/CustomCalender";
import {
  PaginationStyle,
  SelectedTimeLine,
  TimeLinePagination,
  TimeLineStyleContainer,
} from "./TimeLineComponentStyles";
import dayjs from "dayjs";

interface IProps {
  currentSnapshot: ISnapshot;
  snapshotList: ISnapshot[];
  snapshotHandler: (snapshotData: ISnapshot) => void;
}

const TimeLineComponent: React.FC<IProps> = ({
  currentSnapshot,
  snapshotList,
  snapshotHandler,
}) => {
  const [bottomNav, setBottomNav] = useState(false);
  const [page, setPage] = useState(2);
  const [oldDate, setOldDate] = useState("");
  const [newDate, setNewDate] = useState("");
  const handleChange = (event: React.ChangeEvent<unknown>, value: number) => {
    setPage(value);
  };

  const toggleTimeline = () => {
    setBottomNav(!bottomNav);
  };

  const setCurrentSnapshot = (snapshot: ISnapshot) => {
    snapshotHandler(snapshot);
  };

  // const getSnapshotDate = () => {
  //   if (currentSnapshot) {
  //     return Moment(currentSnapshot.date).format("Do MMM YYYY");
  //   } else {
  //     return "No Reality";
  //   }
  // };

  useEffect(() => {
    setCurrentSnapshot(snapshotList[page - 1]);
  }, [page]);

  useEffect(() => {
    if (snapshotList.length > 0) {
      setOldDate(snapshotList[0].date);
      setNewDate(snapshotList[snapshotList.length - 1].date);
      setPage(snapshotList.length);
    }
  }, [snapshotList]);

  const disableWeekends = (date: any) => {
    const timelineDates: any[] = [];
    if (snapshotList.length) {
      snapshotList.forEach((element, i) => {
        timelineDates.push(dayjs(element.date).format("YYYY-MM-DD"));
      });
    }
    return timelineDates.indexOf(dayjs(date).format("YYYY-MM-DD")) < 0;
  };

  const calenderStyles = {
    sx: {
      "& .css-bkrceb-MuiButtonBase-root-MuiPickersDay-root:not(.Mui-disabled,.Mui-selected)":
        {
          backgroundColor: "rgba(0, 0, 0, 0.20)",
        },
    },
  };

  // console.log(snapshotList, "snaphsot listt");
  return (
    <>
      {snapshotList && snapshotList.length > 0 && (
        <TimeLineStyleContainer>
          <SelectedTimeLine
            style={{ bottom: bottomNav ? "" : 0 }}
            onClick={toggleTimeline}
            data-testid={"selected-timeline"}
          >
            {Moment(currentSnapshot?.date).format("DD MMM YYYY")}
          </SelectedTimeLine>

          {bottomNav ? (
            <div
              data-testid="bottomNav"
              //  className="absolute flex flex-col items-center z-10 top-0 inset-x-0"
            >
              <div
              // className="bg-gray-300 border border-gray-700 rounded duration-300 cursor-pointer"
              >
                {/* <p onClick={toggleTimeline}>{getSnapshotDate()}</p> */}
              </div>
              {snapshotList && currentSnapshot && (
                <TimeLinePagination>
                  <div className=" flex text-sm">
                    <p>{oldDate && Moment(oldDate).format("DD MMM YY")}</p>
                  </div>
                  <PaginationStyle
                    count={snapshotList?.length}
                    page={page}
                    onChange={handleChange}
                    data-testid={"page"}
                    renderItem={(item: any) => (
                      <PaginationItem
                        slots={{
                          previous: KeyboardDoubleArrowLeftIcon,
                          next: KeyboardDoubleArrowRightIcon,
                        }}
                        {...item}
                      />
                    )}
                  />
                  <div className="flex text-sm items-center ml-1 ">
                    <p>{newDate && Moment(newDate).format("DD MMM YY")} </p>
                  </div>
                  {!isNaN(page) ? (
                    <CustomCalender
                      onChange={(e: any) => handleChange(e, 1)}
                      data-testid="calender"
                      shouldDisableDate={disableWeekends}
                      hideTextField
                      data={{
                        disableAll: true,
                        defaultValue: currentSnapshot?.date,
                        // defaultValue: page === 2 ? newDate : oldDate,
                        disableDays: disableWeekends,
                        styles: calenderStyles,
                      }}
                    />
                  ) : null}
                </TimeLinePagination>
              )}
            </div>
          ) : null}
        </TimeLineStyleContainer>
      )}
    </>
  );
};

export default TimeLineComponent;
