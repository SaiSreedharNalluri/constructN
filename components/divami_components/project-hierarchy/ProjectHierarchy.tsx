import AddIcon from "@mui/icons-material/Add";
import RemoveIcon from "@mui/icons-material/Remove";
import { TreeItem, TreeView } from "@mui/lab";
import { useEffect, useState } from "react";
import { ChildrenEntity } from "../../../models/IStructure";
import closeIcon from "../../../public/images/closeIcon.svg";
import SearchImg from "../../../public/images/search.svg";
import projectHierIcon from "../../../public/divami_icons/projectHierIcon.svg";
import {
  CloseIcon,
  HeaderLabel,
  HeaderLabelContainer,
  ProjectHierarchyContainer,
  SearchContainer,
  StyledTreeItem,
  TreeViewContainer,
  StyledTreeView,
  CustomInputField,
  MessageContainer,
  ErrorImageDiv,
  ImageErrorIcon,
  MessageDivShowErr,
  LabelContainer,
  // useStyles,
  StyledSpan,
  LabelIcon,
  LabelText,
} from "./StyledComponents";
import type {
  ProjectHierarchyProps,
  RenderTree,
  SelectLayerProps,
} from "./Type";
import { getAllIds, getSelectedLayers } from "./Utils";
import Image from "next/image";

const ProjectHierarchy = ({
  title,
  openSelectLayer,
  onCloseHandler,
  treeData,
  getStructureData,
  handleSearch,
  handleNodeSelection,
  selectedNodes,
  handleNodeExpand,
  expandedNodes,
  setHierarchy,
}: ProjectHierarchyProps) => {
  const [treeViewData, setTreeViewData] = useState<ChildrenEntity[]>(treeData);

  const [block, setBlock] = useState(treeData);

  const [selectedLayers, setSelectedLayers] = useState<string[] | null>(null);

  const handleExpand = () => {
    handleNodeExpand(getAllIds(treeViewData));
  };
  console.log(expandedNodes, "fsffs");
  useEffect(() => {
    // console.log(treeData, "Robn");

    setTreeViewData(treeData);
  }, [treeData.length]);

  console.log("treeViewData22", treeViewData);

  // useEffect(() => {
  //   if (window.localStorage.getItem("nodeData") && getStructureData) {
  //     let nodeData = JSON.parse(window.localStorage.getItem("nodeData") || "");
  //     console.log("nodeData", nodeData);
  //     // if (nodeData && getStructureData) {
  //     //   getStructureData(nodeData);
  //     // }
  //   }
  // }, [treeViewData]);

  useEffect(() => {
    if (window.localStorage.getItem("nodeData")) {
      let nodeData = JSON.parse(window.localStorage.getItem("nodeData") || "");
      if (nodeData && getStructureData) {
        getStructureData(nodeData);
      }
    }
  }, [treeViewData]);

  const onLabelClick = (event: any, nodes: any) => {
    {
      console.log("onclickinlable");
      window.localStorage.setItem("nodeData", JSON.stringify(nodes));
      getStructureData ? getStructureData(nodes) : null;

      if (
        !(
          nodes.children &&
          Array.isArray(nodes.children) &&
          nodes.children.length
        )
      ) {
        setHierarchy(false);
      }
    }
  };

  const renderTreeNode = (node: ChildrenEntity, onLabelClick: any) => (
    <LabelContainer data-testid="label">
      <LabelText onClick={(e: any) => onLabelClick(e, node)}>
        {node.name}
      </LabelText>
      {node.children?.length ? (
        expandedNodes.includes(node._id) ? (
          <LabelIcon
            onClick={(e: any) => {
              handleNodeExpand(getAllIds(node));
            }}
          >
            <RemoveIcon
              style={{ cursor: "pointer", fontSize: "14px" }}
              data-testid={"addIcon"}
            />
          </LabelIcon>
        ) : (
          <StyledSpan onClick={(e) => handleNodeExpand(getAllIds(node))}>
            <AddIcon
              style={{ cursor: "pointer", fontSize: "14px" }}
              data-testid={"addIcon"}
            />
          </StyledSpan>
        )
      ) : (
        <></>
      )}
    </LabelContainer>
  );

  const [search, setSearch] = useState(false);
  const [searchField, setSearchField] = useState("");
  // const [filterBlock, setFilterBlock] = useState<any>(block);

  console.log("searchFieldString", searchField);
  console.log("setBlock", block);

  const handleSearchResult = (e: any) => {
    console.log(e);
    let returnedTree = handleSearch(e);
    // console.log("robn", returnedTree)
    setSearch(true);
    setTreeViewData(returnedTree);
  };

  const renderTree = (nodes: ChildrenEntity, onLabelClick: any) => (
    <>
      <StyledTreeItem
        // needClick={nodes?.children && nodes?.children?.length > 0 ? false : true}
        key={nodes._id}
        nodeId={nodes._id}
        label={renderTreeNode(nodes, onLabelClick)}
        data-testid={"treeItem"}
        style={{ borderBottom: "1px solid #D9D9D9" }}
      >
        {Array.isArray(nodes.children) && nodes.children.length
          ? nodes.children.map((node) => renderTree(node, onLabelClick))
          : null}
      </StyledTreeItem>
    </>
  );

  useEffect(() => {
    const layersSelected = getSelectedLayers(treeViewData);
    setSelectedLayers(layersSelected);
    console.log([...layersSelected], "selectedLayers");
    console.log(search);
    console.log(selectedNodes);

    search ? handleExpand() : null;
  }, [treeViewData]);

  // useEffect(() => {
  //   const newFilteredBlocklist = block[0]?.children?.filter((val: any) => {
  //     // return val;
  //     // console.log(val.name.toLocaleLowerCase());
  //     return val?.name?.toLocaleLowerCase().includes(searchField);

  //     console.log("effect firingn");
  //   });
  //   setFilterBlock(newFilteredBlocklist);
  // }, [block, searchField]);

  const handleToggle = (event: React.SyntheticEvent, nodeIds: string[]) => {
    console.log(nodeIds, "nodeIds");
    handleNodeExpand(nodeIds);
  };

  const handleSelect = (event: React.SyntheticEvent, nodeIds: string[]) => {
    handleNodeSelection(nodeIds);
  };

  const onSearchChange = (event: any) => {
    let parentArr = [...treeData];
    const searchFieldString = event.target.value.toLocaleLowerCase();
    let newObj = [
      ...parentArr[0]?.children?.filter(
        (item: any, index: number) =>
          item.name.toLocaleLowerCase().includes(searchFieldString)
        //     || item.name.children.filter((item:any,index:number) =>)
      ),
    ];
    console.log("newobjj", newObj);
    parentArr = [{ ...parentArr[0], children: [...newObj] }];
    //  parentObj[0].children = newObj
    setTreeViewData([...parentArr]);
    console.log(
      "Robby",
      searchFieldString,
      treeData[0].children.length,
      parentArr[0].children.length
    );
  };

  // console.log("hiiii", treeViewData, filterBlock);

  return (
    <ProjectHierarchyContainer>
      <HeaderLabelContainer>
        <HeaderLabel style={{ color: "#101F4B", textAlign: "center" }}>
          {title}
        </HeaderLabel>
        <CloseIcon
          src={closeIcon}
          onClick={onCloseHandler}
          alt={"close Icon"}
        />
      </HeaderLabelContainer>
      <SearchContainer>
        <CustomInputField
          id={"search"}
          variant="outlined"
          autoComplete="off"
          placeholder={"Search"}
          // onChange={(e: any) => {
          //   onSearchChange(e);
          // }}

          onChange={(e: any) => {
            handleSearchResult(e);
          }}
          data-testid={"search"}
          // InputProps={{
          //   // ...params.InputProps,
          //   startAdornment: (
          //     <InputAdornment position="start">
          //       <SearchIcon src={SearchIcon} alt="search" />
          //     </InputAdornment>
          //   ),
          // }}
          InputProps={{
            startAdornment: <Image src={SearchImg} alt="search" />,
            // ariaLabel:{"search-input"}
          }}
        />
      </SearchContainer>
      <TreeViewContainer
        style={{ overflow: "auto", height: `calc(100vh - 300px)` }}
      >
        {treeViewData.length === 0 ? (
          <ErrorImageDiv>
            <ImageErrorIcon src={projectHierIcon} alt="Error Image" />
            <MessageDivShowErr>No result found</MessageDivShowErr>
          </ErrorImageDiv>
        ) : (
          <StyledTreeView
            aria-label="rich object"
            defaultCollapseIcon={<></>}
            defaultExpandIcon={<></>}
            expanded={expandedNodes}
            selected={selectedNodes}
            onNodeToggle={handleToggle}
            onNodeSelect={handleSelect}
          >
            {treeViewData.map((eachNode) => renderTree(eachNode, onLabelClick))}
          </StyledTreeView>
        )}
      </TreeViewContainer>
    </ProjectHierarchyContainer>
  );
};

export default ProjectHierarchy;
