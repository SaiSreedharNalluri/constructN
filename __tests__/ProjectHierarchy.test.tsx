
import TimeLineComponent from "../components/divami_components/timeline-container/TimeLineComponent";
import { fireEvent, render, screen } from "@testing-library/react";
import ProjectHierarchy from "../components/divami_components/project-hierarchy/ProjectHierarchy";
import '@testing-library/jest-dom';
import { getAllIds, getSelectedLayers, handleSelection } from "../components/divami_components/project-hierarchy/Utils";
import { ChildrenEntity } from "../models/IStructure";
import { RenderTree } from "../components/divami_components/project-hierarchy/Type";
const treeData = [
    {
        "_id": "STR468587",
        "name": "Adani Koregaon Park",
        "type": "Exterior",
        "isExterior": true,
        "project": "PRJ201897",
        "parent": null,
        "children": [
            {
                "_id": "STR536138",
                "name": "A3_Block",
                "type": "Exterior",
                "isExterior": true,
                "project": "PRJ201897",
                "parent": "STR468587",
                "children": [],
                "location": [],
                "designs": [
                    {
                        "_id": "DSG234293",
                        "type": "BIM",
                        "name": "adhani_a3_block.rvt",
                        "project": "PRJ201897",
                        "structure": "STR536138",
                        "storage": [
                            {
                                "provider": "constructn-oss",
                                "path": "PRJ201897/structures/STR536138/designs/DSG234293/adhani_a3_block.rvt",
                                "providerType": "internal",
                                "format": ".RVT"
                            },
                            {
                                "provider": "autodesk-oss",
                                "pathId": "dXJuOmFkc2sub2JqZWN0czpvcy5vYmplY3Q6YWRoYW5pX2tvcmVnYW9uX3BhcmtfcHVuZS8lMkZQUkoyMDE4OTclMkZzdHJ1Y3R1cmVzJTJGU1RSNTM2MTM4JTJGZGVzaWducyUyRkRTRzIzNDI5MyUyRmFkaGFuaV9hM19ibG9jay5ydnQ",
                                "providerType": "external",
                                "format": ".SVF2"
                            }
                        ],
                        "createdAt": "2023-02-09T17:12:52.952Z",
                        "updatedAt": "2023-02-09T17:12:52.952Z",
                        "__v": 0
                    }
                ]
            },
            {
                "_id": "STR511876",
                "name": "A2_Block",
                "type": "Exterior",
                "isExterior": true,
                "project": "PRJ201897",
                "parent": "STR468587",
                "children": [{
                    "_id": "STR780049",
                    "name": "1_Floor_A2_Block",
                    "type": "Interior",
                    "isExterior": false,
                    "project": "PRJ201897",
                    "parent": "STR511876",
                    "children": [],
                    "location": [],
                    "designs": [
                        {
                            "_id": "DSG051690",
                            "type": "Plan Drawings",
                            "name": "floormap.pdf",
                            "project": "PRJ201897",
                            "structure": "STR780049",
                            "storage": [
                                {
                                    "provider": "constructn-oss",
                                    "path": "PRJ201897/structures/STR780049/designs/DSG051690/floormap.pdf",
                                    "providerType": "internal",
                                    "format": ".PDF"
                                },
                                {
                                    "provider": "autodesk-oss",
                                    "pathId": "dXJuOmFkc2sub2JqZWN0czpvcy5vYmplY3Q6YWRoYW5pX2tvcmVnYW9uX3BhcmtfcHVuZS8lMkZQUkoyMDE4OTclMkZzdHJ1Y3R1cmVzJTJGU1RSNzgwMDQ5JTJGZGVzaWducyUyRkRTRzA1MTY5MCUyRkEyLUYxLnBkZg",
                                    "providerType": "external",
                                    "format": ".SVF2"
                                }
                            ],
                            "createdAt": "2023-02-09T12:42:19.836Z",
                            "updatedAt": "2023-02-09T12:42:19.836Z",
                            "__v": 0
                        },
                        {
                            "_id": "DSG196412",
                            "type": "BIM",
                            "name": "Block A2.rvt",
                            "project": "PRJ201897",
                            "structure": "STR780049",
                            "storage": [
                                {
                                    "provider": "constructn-oss",
                                    "path": "PRJ201897/structures/STR780049/designs/DSG196412/Block A2.rvt",
                                    "providerType": "internal",
                                    "format": ".RVT"
                                },
                                {
                                    "provider": "autodesk-oss",
                                    "pathId": "dXJuOmFkc2sub2JqZWN0czpvcy5vYmplY3Q6YWRoYW5pX2tvcmVnYW9uX3BhcmtfcHVuZS8lMkZQUkoyMDE4OTclMkZzdHJ1Y3R1cmVzJTJGU1RSNTExODc2JTJGZGVzaWducyUyRkRTRzI4MjkwNCUyRmFkaGFuaV9hMl9ibG9jay5ydnQ",
                                    "providerType": "external",
                                    "format": ".SVF2"
                                }
                            ],
                            "createdAt": "2023-02-09T12:42:19.836Z",
                            "updatedAt": "2023-02-09T12:42:19.836Z",
                            "__v": 0
                        }
                    ]
                },],
                "location": [],
                "designs": [
                    {
                        "_id": "DSG282904",
                        "type": "BIM",
                        "name": "adhani_a2_block.rvt",
                        "project": "PRJ201897",
                        "structure": "STR511876",
                        "storage": [
                            {
                                "provider": "constructn-oss",
                                "path": "PRJ201897/structures/STR511876/designs/DSG282904/adhani_a2_block.rvt",
                                "providerType": "internal",
                                "format": ".RVT"
                            }
                        ],
                        "createdAt": "2023-02-10T06:09:05.196Z",
                        "updatedAt": "2023-02-10T06:09:05.196Z",
                        "__v": 0
                    }
                ]
            },
            {
                "_id": "STR550768",
                "name": "C_Block",
                "type": "Exterior",
                "isExterior": true,
                "project": "PRJ201897",
                "parent": "STR468587",
                "children": [{
                    "_id": "STR132657",
                    "name": "1_Floor_C_Block",
                    "type": "Interior",
                    "isExterior": false,
                    "project": "PRJ201897",
                    "parent": "STR550768",
                    "children": [],
                    "location": [],
                    "designs": [
                        {
                            "_id": "DSG269650",
                            "type": "Plan Drawings",
                            "name": "floormap.pdf",
                            "project": "PRJ201897",
                            "structure": "STR132657",
                            "storage": [
                                {
                                    "provider": "constructn-oss",
                                    "path": "PRJ201897/structures/STR132657/designs/DSG269650/floormap.pdf",
                                    "providerType": "internal",
                                    "format": ".PDF"
                                },
                                {
                                    "provider": "autodesk-oss",
                                    "pathId": "dXJuOmFkc2sub2JqZWN0czpvcy5vYmplY3Q6YWRoYW5pX2tvcmVnYW9uX3BhcmtfcHVuZS8lMkZQUkoyMDE4OTclMkZzdHJ1Y3R1cmVzJTJGU1RSMTMyNjU3JTJGZGVzaWducyUyRkRTRzI2OTY1MCUyRkMtRjEucGRm",
                                    "providerType": "external",
                                    "format": ".SVF2"
                                }
                            ],
                            "createdAt": "2023-02-09T12:48:07.420Z",
                            "updatedAt": "2023-02-09T12:48:07.420Z",
                            "__v": 0
                        },
                        {
                            "_id": "DSG239842",
                            "type": "BIM",
                            "name": "Block C.rvt",
                            "project": "PRJ201897",
                            "structure": "STR132657",
                            "storage": [
                                {
                                    "provider": "constructn-oss",
                                    "path": "PRJ201897/structures/STR132657/designs/DSG239842/Block C.rvt",
                                    "providerType": "internal",
                                    "format": ".RVT"
                                },
                                {
                                    "provider": "autodesk-oss",
                                    "pathId": "dXJuOmFkc2sub2JqZWN0czpvcy5vYmplY3Q6YWRoYW5pX2tvcmVnYW9uX3BhcmtfcHVuZS8lMkZQUkoyMDE4OTclMkZzdHJ1Y3R1cmVzJTJGU1RSNTUwNzY4JTJGZGVzaWducyUyRkRTRzMwMjM4MyUyRmFkaGFuaV9jX2Jsb2NrLnJ2dA",
                                    "providerType": "external",
                                    "format": ".SVF2"
                                }
                            ],
                            "createdAt": "2023-02-09T12:48:07.420Z",
                            "updatedAt": "2023-02-09T12:48:07.420Z",
                            "__v": 0
                        }
                    ]
                },],
                "location": [],
                "designs": [
                    {
                        "_id": "DSG302383",
                        "type": "BIM",
                        "name": "adhani_c_block.rvt",
                        "project": "PRJ201897",
                        "structure": "STR550768",
                        "storage": [
                            {
                                "provider": "constructn-oss",
                                "path": "PRJ201897/structures/STR550768/designs/DSG302383/adhani_c_block.rvt",
                                "providerType": "internal",
                                "format": ".RVT"
                            }
                        ],
                        "createdAt": "2023-02-10T06:09:42.783Z",
                        "updatedAt": "2023-02-10T06:09:42.783Z",
                        "__v": 0
                    }
                ]
            }
        ],
        "location": [],
        "designs": []
    }
]
beforeEach(() => {
    const container = render(
        <ProjectHierarchy
            handleSearch={(event: React.ChangeEvent<HTMLInputElement>) => {}}
            title={"Project Hierarchy"}
            onCloseHandler={() => {}}
            treeData={treeData}
            setHierarchy={() => {}}
            getStructureData={() => {}}
            handleNodeSelection={() => {}}
            selectedNodes={() => {}}
            handleNodeExpand={() => {}}
            expandedNodes={['STR468587']}
          />
      );  
});

it("renders component unchanged", () => {
  const { container } = 
    render(

        <ProjectHierarchy
            handleSearch={(event: React.ChangeEvent<HTMLInputElement>) => {}}
            title={"Project Hierarchy"}
            onCloseHandler={jest.fn()}
            treeData={treeData}
            setHierarchy={true}
            getStructureData={jest.fn()}
            handleNodeSelection={jest.fn()}
            selectedNodes={jest.fn()}
            handleNodeExpand={jest.fn()}
            expandedNodes={[]}
          />
      ); 
  expect(container).toMatchSnapshot();
});
it("should check search functionality", () => {
    const search = screen.getAllByPlaceholderText('Search');
    fireEvent.change(search[0],{target:{value:'A'}});
    expect(search[0]).toBeInTheDocument()
})

it("should check expand functionality", () => { 
    const search = screen.getAllByTestId("addIcon")
    fireEvent.click(document.getElementsByClassName('MuiTreeItem-iconContainer')[0])
    expect(screen.getByText('A3_Block')).toBeInTheDocument()
})

it("should check getStructure functionality", () => { 
    fireEvent.click(document.getElementsByClassName('MuiTreeItem-iconContainer')[0])
    fireEvent.click(screen.getByText('A3_Block'))
    expect(screen.getByText('A3_Block')).toBeInTheDocument()
})

it("should check the getAllId's function", () => {
    const allIds = getAllIds(treeData)
    expect(allIds).toEqual(["STR468587","STR536138","STR511876","STR780049","STR550768","STR132657",])
})

it("should check getSelectedLayers function", () => {
    const selected = getSelectedLayers(treeData as unknown as ChildrenEntity[])
    expect(selected).toEqual([])
})

it("should check handleSelection function",() => {
    const selected = handleSelection(treeData as unknown as RenderTree[],'STR132657')
    expect(selected).not.toBeNull()
})
