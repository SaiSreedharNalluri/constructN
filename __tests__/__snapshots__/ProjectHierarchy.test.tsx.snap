// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component unchanged 1`] = `
<div>
  <div
    class="css-jg9knl"
  >
    <div
      class="css-1bdfr09"
    >
      <div
        class="css-pqcd7p"
        style="color: rgb(16, 31, 75); text-align: center;"
      >
        Project Hierarchy
      </div>
      <img
        alt="close Icon"
        class="css-4g6ai3"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </div>
    <div
      class="css-7r3r5t"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1s2z0n5-MuiFormControl-root-MuiTextField-root"
        data-testid="search"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <img
            alt="search"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id="search"
            placeholder="Search"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="css-1hlpyp9"
      style="overflow: auto; height: 63vh;"
    >
      <ul
        aria-label="rich object"
        aria-multiselectable="false"
        class="MuiTreeView-root css-vbuvlb-MuiTreeView-root"
        id=":r3:"
        role="tree"
        tabindex="0"
      >
        <li
          aria-expanded="false"
          class="MuiTreeItem-root css-28ckyw-MuiTreeItem-root"
          data-testid="treeItem"
          id=":r3:-STR468587"
          role="treeitem"
          style="border-bottom: 1px solid #D9D9D9;"
          tabindex="-1"
        >
          <div
            class="css-1g86id8-MuiTreeItem-content MuiTreeItem-content"
          >
            <div
              class="MuiTreeItem-iconContainer"
            />
            <div
              class="MuiTreeItem-label"
            >
              <div
                class="css-gg4vpm"
                data-testid="label"
              >
                <span>
                  Adani Koregaon Park
                </span>
                <span
                  class="css-1nd4bm0"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="addIcon"
                    focusable="false"
                    style="cursor: pointer; font-size: 14px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
